File: day1.py
import os

def read_file(file):
    with open(file) as f:
        #print(f)
        lines = f.readlines()
        #print(lines)
    return [line.strip("\n") for line in lines] 


def is_larger(input):
    count = 0
    for i in range(1,len(input)):
        if int(input[i]) > int(input[i-1]):
            count += 1
    return count


def count_threes(input):
    counts = []
    for i in range(len(input) -2):
        counts.append( int(input[i]) + int(input[i + 1]) + int(input[i + 2]))
    return counts



if __name__ == "__main__":
    input = read_file(os.getcwd() + "\input\day1.txt")
    threes = count_threes(input)
    result = is_larger(threes)
    print(result)

File: day1.hs
import System.IO

-- runs second part
main :: IO Int
main = do
        contents <- readFile "input/day1.txt"
        let intList = map (read::String->Int) (lines contents)
        return (doThreeThing intList)

-- does the first part
doThing :: [Int] -> Int
doThing [] = 0
doThing [x] = 0
doThing [x, y]  | x < y = 1
                | otherwise = 0

doThing (x:y:xs)    | x < y = 1 + doThing (y:xs)
                    | otherwise = 0 + doThing(y:xs)


-- does second part
doThreeThing :: [Int] -> Int
doThreeThing [] = 0
doThreeThing [_] = 0
doThreeThing [x, y, z, x1]  | x+y+z < y+z+x1 = 1
                            | otherwise = 0

doThreeThing (x:y:z:x1:xs)  | x+y+z < y+z+x1 = 1 + doThreeThing (y:z:x1:xs)
                            | otherwise = 0 + doThreeThing(y:z:x1:xs)

